shell/utils/auth.js

```diff
@@ -187,9 +187,7 @@ export function setProduct(store, to) {
   (product && !store.getters['type-map/isProductRegistered'](product))) {
     const error = new Error(store.getters['i18n/t']('nav.failWhale.productNotFound', { productNotFound: product }, true));
 
-    store.dispatch('loadingError', error);
-
-    throw new Error('loadingError', new Error(store.getters['i18n/t']('nav.failWhale.productNotFound', { productNotFound: product }, true)));

```

Comment:
This was getting caught somewhere in authenticated middleware. We no longer need to do this. The 'loadingError' store action is sufficient.

You can verify this by trying to load an unknown product i.e. `/c/local/asdasdasdasda`

Reply:
I've seen the same stuff in the entry initialization. Are the 2 connected or what? Because I see 2 different errors triggered. Are these just 2 separated phases: access and authentication?

Reply:
My best guess is that at one point `store.dispatch('loadingError', error);` was sufficient. I think the addition of the thrown exception was to force the consumers to exit early.